client/components/navbar.jsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+
+export default class Navbar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      search: false,
+      searchValue: ''
+    };
+    this.handleSearchClick = this.handleSearchClick.bind(this);
+    this.handleCancelClick = this.handleCancelClick.bind(this);
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleSearchSubmit = this.handleSearchSubmit.bind(this);
+  }
+
+  handleSearchClick() {
+    this.setState({ search: !this.state.search });
+  }
+
+  handleCancelClick() {
+    this.setState({
+      search: false,
+      searchValue: ''
+    });
+  }
+
+  handleInputChange(event) {
+    this.setState({ searchValue: event.target.value });
+  }
+
+  handleSearchSubmit(event) {
+    event.preventDefault();
+    const query = this.state.searchValue.replaceAll(' ', '+');
+    const url = new URL(window.location);
+    url.hash = ('search?q=' + query);
+    window.location.replace(url);
```

Comment:
I recommend using assignment instead, typically searches add to the browser history.

```js
window.location.hash = `search?q=${query}`;
```